### PR TITLE
Add support to qobj_to_circuits for qobj without qasm

### LINKED ISSUE #1407: qobj_to_circuits not using the compiled qasm in the qobj header
### Informations

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit adds support to the helper function in
qiskit.tools.qobj_to_circuits for creating circuits from a qobj without
compiled qasm in the header. It does this by looping over the Qobj
object to run the operations needed to construct the circuit objects
directly instead of just parsing qasm.

### Details and comments

Fixes #1407

### CODE REVIEW

File: qiskit/tools/qobj_to_circuits.py [hunk 1]
```diff
@@ -23,8 +39,35 @@ def qobj_to_circuits(qobj):
         for x in qobj.experiments:
             if hasattr(x.header, 'compiled_circuit_qasm'):
                 circuits.append(
-                    qc.QuantumCircuit.from_qasm_str(x.header.compiled_circuit_qasm))
+                    qc.QuantumCircuit.from_qasm_str(
+                        x.header.compiled_circuit_qasm))
+            else:
+                quantum_registers = _get_registers(x.header.qubit_labels,
+                                                   qr.QuantumRegister)
+                classical_registers = _get_registers(x.header.clbit_labels,
```
Comment (ajavadia) [inline qiskit/tools/qobj_to_circuits.py]: I think you can construct the `quantum_registers` and `classical_registers` much more easily if you just look at the `header.qreg_sizes` and `header.creg_sizes` fields.

For example they look like:
```
qreg_sizes=[['qr', 4]]
creg_sizes=[['cr1', 3], ['cr2', 1]]
```
which means:
```
QuantumRegister(size=4, name='qr')
ClassicalRegister(size=3, name='cr1')
ClassicalRegister(size=1, name='cr2')
```

**Comment by jaygambetta** [comment]
can we do this after #1410

File: qiskit/tools/qobj_to_circuits.py [hunk 2]
```diff
@@ -23,8 +39,35 @@ def qobj_to_circuits(qobj):
         for x in qobj.experiments:
             if hasattr(x.header, 'compiled_circuit_qasm'):
                 circuits.append(
-                    qc.QuantumCircuit.from_qasm_str(x.header.compiled_circuit_qasm))
+                    qc.QuantumCircuit.from_qasm_str(
+                        x.header.compiled_circuit_qasm))
+            else:
```
[the remark this answers is not in the document]
Comment (jaygambetta) [reply qiskit/tools/qobj_to_circuits.py]: yeah lets assume it does not exist

[the remark this answers is not in the document]
Comment (mtreinish) [reply qiskit/tools/qobj_to_circuits.py]: Done

File: qiskit/tools/qobj_to_circuits.py (hunk 1 above)
Comment (mtreinish) [reply qiskit/tools/qobj_to_circuits.py]: Ah, good call. That works and is definitely faster than the multiple loops

File: qiskit/converters/qobj_to_circuits.py [hunk 3]
```diff
@@ -6,7 +6,10 @@
 # the LICENSE.txt file in the root directory of this source tree.
 
 """Helper function for converting qobj to a list of circuits"""
-from qiskit.circuit import QuantumCircuit
+
+from qiskit.circuit import classicalregister as cr
+from qiskit.circuit import quantumcircuit as qc
+from qiskit.circuit import quantumregister as qr
```
Comment (jaygambetta) [inline qiskit/converters/qobj_to_circuits.py]: non-blocking but there are no loops so we don't need to import like this.

**Pull request merged**